lib/shared/text-table.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * @fileoverview Optimized version of the `text-table` npm module to improve performance by replacing inefficient regex-based
+ * whitespace trimming with a modern built-in method.
+ *
+ * This modification addresses a performance issue reported in https://github.com/eslint/eslint/issues/18709
+ *
+ * The `text-table` module is published under the MIT License. For the original source, refer to:
+ * https://www.npmjs.com/package/text-table.
+ */
+
+/*
+ * MIT License
+ * Copyright (c) Microsoft Corporation.
```

Comment:
Why Microsoft?

Reply:
Yeah, this is a copy paste error. I'll update

Reply:
Updated.